karen-tracy/scripts/articleView.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+let articleView = {};
+
+// DONE: Where possible, refactor methods into arrow functions, including the document.ready() method at the bottom.
+
+// COMMENT: How do arrow functions affect the context of "this"? How did you determine if a function could be refactored?
+// It changes the scope  to one parent above, so changes the context of this to the parent scope. If there's a contextual this then the arrow function will change the scope and will not work.
+
+articleView.populateFilters = () => {
+
+  let optionTemplateScript = $('#option-template').html();
+  let optionTemplate = Handlebars.compile(optionTemplateScript);
+  let val;
+
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+
+      val = $(this).data('author');
+
+      if ($(`#author-filter option[value="${val}"]`).length === 0) {
+        $(optionTemplate({value:val})).appendTo('#author-filter');
+      }
+
+      val = $(this).data('category');
+
+      if ($(`#category-filter option[value="${val}"]`).length === 0) {
+        $(optionTemplate({value:val})).appendTo('#category-filter');
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = () => {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
```

Comment:
You use this to fade in all of the siblings of the clicked element, which feels brittle to me; it'll get in your way if you ever add more elements as siblings of that clicked link. I would consider finding by a more specific class rather than using the wildcard selector.